src/App.tsx

```diff
@@ -1,9 +1,34 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+type Func = React.Dispatch<React.SetStateAction<string>>;
+
+const debounce = (f: Func,
+  delay: number) => {
+  let timerId: NodeJS.Timeout;
+
+  return (arg: string) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, arg);
+  };
+};
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState(query);
+  const getVisibleMovies = () => (
+    moviesFromServer.filter(movie => (
+      movie.title.toLowerCase().includes(appliedQuery.toLowerCase())
+        || movie.description.toLowerCase().includes(appliedQuery.toLowerCase())
+    ))
```

Comment:
better to create separate variables for `doesTitleMatch / doesDescriptionMatch` in order to improve its readability

Reply:
> Great job with debouncer 🔥 Still I have a couple of comments for you
> 
> * search does not work with spaced query
> 
> <img alt="Screenshot 2022-10-28 at 13 23 17" width="1037" src="https://user-images.githubusercontent.com/103033758/198565322-495903e1-1884-4a2c-91a1-fc59ac8f768c.png">

https://mate-academy.github.io/react_movies-list-filter/ - we can watch on "working version" and spaces behavior there.